builtin/providers/aws/resource_aws_internet_gateway.go

```diff
@@ -46,18 +46,40 @@ func resourceAwsInternetGatewayCreate(d *schema.ResourceData, meta interface{})
 	log.Printf("[INFO] InternetGateway ID: %s", d.Id())
 
 	resource.Retry(5*time.Minute, func() *resource.RetryError {
+		log.Printf("[INFO] Refreshing InternetGateway ID: %s", d.Id())
 		igRaw, _, err := IGStateRefreshFunc(conn, d.Id())()
 		if igRaw != nil {
 			return nil
 		}
 		if err == nil {
+			log.Printf("[INFO] RetryableError creating InternetGateway ID: %s %s", d.Id(), err)
 			return resource.RetryableError(err)
 		} else {
+			log.Printf("[INFO] NonRetryableError creating InternetGateway ID: %s %s", d.Id(), err)
 			return resource.NonRetryableError(err)
 		}
 	})
 
-	err = setTags(conn, d)
+	err = resource.Retry(5*time.Minute, func() *resource.RetryError {
+		log.Printf("[INFO] Setting tags InternetGateway ID: %s", d.Id())
+		err = setTags(conn, d)
+		if err == nil {
+			return nil
+		}
+		ec2err, ok := err.(awserr.Error)
+		if !ok {
+			log.Printf("[INFO] RetryableError setting tags InternetGateway ID: %s %s", d.Id(), err)
+			return resource.RetryableError(err)
```

Comment:
Seems like most of your retries follow this pattern, are you assuming this situation arrises outside of AWS, and so should be retried?